bbq/compiler/compiler.go

```diff
@@ -931,13 +931,7 @@ func (c *Compiler[_]) VisitNilExpression(_ *ast.NilExpression) (_ struct{}) {
 func (c *Compiler[_]) VisitIntegerExpression(expression *ast.IntegerExpression) (_ struct{}) {
 	integerType := c.ExtendedElaboration.IntegerExpressionType(expression)
 	constantKind := constantkind.FromSemaType(integerType)
-
-	// TODO:
-	var data []byte
-	data = leb128.AppendInt64(data, expression.Value.Int64())
-
-	constant := c.addConstant(constantKind, data)
-	c.codeGen.Emit(opcode.InstructionGetConstant{ConstantIndex: constant.index})
+	c.intConstLoad(constantKind, expression.Value.Int64())
```

Comment:
Small nit: Might be good to leave the TODO here, because we still need to support larger types in the future. In any case, added it to #3769, so we don't forget